lti_consumer/plugin/views.py

```diff
@@ -231,3 +235,80 @@ def scores(self, request, *args, **kwargs):
             status=status.HTTP_201_CREATED,
             headers=headers
         )
+
+
+class LtiNrpsContextMembershipViewSet(viewsets.ModelViewSet):
+    """
+    LTI NRPS Context Membership Service endpoint.
+    """
+
+    # Custom permission classes for LTI APIs
+    authentication_classes = [Lti1p3ApiAuthentication]
+    permission_classes = [LtiNrpsContextMembershipsPermissions]
+
+    # Renderer classes to accept LTI NRPS content types
+    renderer_classes = [
+        MembershipResultRenderer,
+    ]
+
+    serializer_class = LtiNrpsContextMembershipSerializer
+
+    # Custom pagination class to support pagination via Link Header
+    pagination_class = LinkHeaderPagination
+
+    def parse_role_filter(self):
+        """
+        Parse role filter from query param and map it to edx role.
+        """
+        role_filter_param = self.request.query_params.get('role', None)
+
+        matched_role = None
+        if role_filter_param:
+            for edx_role, context_roles in LTI_1P3_CONTEXT_ROLE_MAP.items():
+                if role_filter_param in context_roles:
+                    matched_role = edx_role
+
+        return matched_role
+
+    def get_queryset(self):
+        """
+        Get User queryset related to current course. Also filter via role if provided.
+        """
+
+        # get course key
+        course_key = self.request.lti_configuration.location.course_key
+
+        kwargs = {
+            'access_roles': [],
+            'include_students': True,
+            'prefetch_accessroles': True,
+        }
+
+        # check if role filter has given
+        filter_by = self.parse_role_filter()
+
+        if filter_by is None:
+            # if no role filter given, we are interested in instructor & staff. other access
+            # roles like beta tester etc doesn't have equivalent role in LTI Context Role vocab.
+            kwargs['access_roles'] = ['instructor', 'staff']
+        elif filter_by != 'student':
+            kwargs['access_roles'] = [filter_by]
+            # exclude students if filtered by any access role
+            kwargs['include_students'] = False
```

Comment:
Is there any use case where we want to exclude students? If so, can't we do it in the same way as staff or instructor?

Reply:
AFAIK, in ``edx-platform``, ``student`` is not a ``CourseAccessRole``. As this API is part of ``edx-platform``, I thought this might make more sense to treat ``access_roles`` as ``CourseAccessRole`` and students as ``CourseEnrollment``. And the consumer of the API (lti_consumer xblock in this case) would build the required parameters themselves.